azure/durable_functions/models/TaskOrchestrationExecutor.py

```diff
@@ -0,0 +1,280 @@
+from azure.durable_functions.models.NewTask import TaskBase, TaskState, AtomicTask
+from azure.durable_functions.models.OrchestratorState import OrchestratorState
+from azure.durable_functions.models.DurableOrchestrationContext import DurableOrchestrationContext
+from typing import Any, List, Optional
+from azure.durable_functions.models.history.HistoryEventType import HistoryEventType
+from azure.durable_functions.models.history.HistoryEvent import HistoryEvent
+from types import GeneratorType
+from collections import namedtuple
+import json
+from ..models.entities.ResponseMessage import ResponseMessage
+from azure.functions._durable_functions import _deserialize_custom_object
+
+
+class TaskOrchestrationExecutor:
+    """Manages the execution and replay of user-defined orchestrations."""
+
+    def __init__(self):
+        """Initialize TaskOrchestrationExecutor."""
+        # A mapping of event types to a tuple of
+        #   (1) whether the event type represents a task success
+        #   (2) the attribute in the corresponding event object that identifies the Task
+        # this mapping is used for processing events that transition a Task from its running state
+        # to a terminal one
+        SetTaskValuePayload = namedtuple("SetTaskValuePayload", ("is_success", "task_id_key"))
+        self.event_to_SetTaskValuePayload = dict([
+            (HistoryEventType.TASK_COMPLETED, SetTaskValuePayload(True, "TaskScheduledId")),
+            (HistoryEventType.TIMER_FIRED, SetTaskValuePayload(True, "TimerId")),
+            (HistoryEventType.SUB_ORCHESTRATION_INSTANCE_COMPLETED,
+             SetTaskValuePayload(True, "TaskScheduledId")),
+            (HistoryEventType.EVENT_RAISED, SetTaskValuePayload(True, "Name")),
+            (HistoryEventType.TASK_FAILED, SetTaskValuePayload(False, "TaskScheduledId")),
+            (HistoryEventType.SUB_ORCHESTRATION_INSTANCE_FAILED,
+             SetTaskValuePayload(False, "TaskScheduledId"))
+        ])
+        self.task_completion_events = set(self.event_to_SetTaskValuePayload.keys())
+        self.initialize()
+
+    def initialize(self):
+        """Initialize the TaskOrchestrationExecutor for a new orchestration invocation."""
+        # The first task is just a placeholder to kickstart the generator.
+        # So it's value is `None`.
+        # TODO: need to set the `is_replaying` flag in here!
```

Comment:
whoops, seem like I forgot about this `TODO`. I'll tacke this tomorrow as well

Reply:
As note for myself, I should also review that `is_replaying` is properly updated for compound tasks.